Source/Schema.NET/ValuesConverter.cs

```diff
@@ -219,6 +219,22 @@ public virtual void WriteObject(JsonWriter writer, object value, JsonSerializer
         private static string GetTypeNameFromToken(JToken token)
         {
             var typeNameToken = token.Values().FirstOrDefault(t => t.Path.EndsWith("@type"));
+            if (typeNameToken is JArray array)
+            {
+                if (!array.Any())
+                {
+                    throw new ArgumentException("Type was an empty array");
```

Comment:
We should try to avoid throwing. It would be better to return a `Thing`.

Reply:
For sure, I can look into it.